### PR TITLE
Feat: Support assets dynamically from the backend

### PR DESCRIPTION
**Pull request by erdimaden**
### What changed? Why?
This makes the SDK able to dynamically support any assets added from the backend.

Main changes

- Fixes balance handling for non-hardcoded assets to convert to whole amounts using the backend specified precision
- Fetches assets before transfer/trade so that we can convert to atomic amounts at the SDK layer without hardcoding

### CODE REVIEW

File: src/coinbase/address.ts [hunk 1]
```diff
@@ -253,19 +272,19 @@ export class Address {
    * Creates a trade model for the specified amount and assets.
    *
    * @param amount - The amount of the Asset to send.
-   * @param fromAssetId - The ID of the Asset to trade from. For Ether, eth, gwei, and wei are supported.
-   * @param toAssetId - The ID of the Asset to trade to. For Ether, eth, gwei, and wei are supported.
+   * @param fromAsset - The Asset to trade from. For Ether, eth, gwei, and wei are supported.
```
Comment (jazz-cb) [inline src/coinbase/address.ts:275]: remove ` For Ether, eth, gwei, and wei are supported.`

File: src/coinbase/asset.ts [hunk 2]
```diff
@@ -1,61 +1,97 @@
...
+      switch (assetId) {
+        case "gwei":
+          decimals = GWEI_DECIMALS;
+          break;
+        case "wei":
+          decimals = 0;
+          break;
+        case "eth":
+          break;
+        default:
```
Comment (jazz-cb) [inline src/coinbase/asset.ts:68]: default should be break. we are supporting more erc20s on mainnet now https://docs-cdp-developer-platform-preview.cbhq.net/wallets/docs/assets/

File: src/coinbase/address.ts (hunk 1 above)
Comment (erdimaden) [reply src/coinbase/address.ts:275]: good catch! updated.

File: src/coinbase/asset.ts (hunk 2 above)
Comment (erdimaden) [reply src/coinbase/asset.ts:68]: this is only for ETH, Wei, or Gwei usage. ERC20 tokens should not be able to bypass the `model.asset_id !== Coinbase.toAssetId(assetId)` guard.

**Review by jazz-cb — APPROVED** [review]

**Pull request merged**